bin/generate_charts.py

```diff
@@ -192,36 +192,22 @@ def create_chart(prefix, workload, title, y_label, time_series):
     chart.render_to_file(f'{output}{workload}.svg')
 
 
-def terminal_print_p99999(title, data):
-    target = list(data.keys())[-2]
-    value = data[target]
-    print(title, target, value)
-
 def generate_charts(files):
 
     aggregate = []
-    topics = 0
-    partitions = 0
-    message_size = 0
-    producers = 0
-    rate = 0
 
+    # Charts are labeled based on benchmark names, we need them
+    # to be unique. A combination of (driver, workload) defines
+    # a unique benchmark and name is "{driver}-{workload}".
+    benchmark_names = set()
     for file in sorted(files):
-        print(file)
         data = json.load(open(file))
-        m = re.search(
-            r'^(\d+)-topic-(\d+)-partitions-(\d+)kb-(\d+)-producers-(\d+)k-rate-(\w+)-.*',
-            path.basename(file))
-        if m:
-            topic = m.group(1)
-            partitions = m.group(2)
-            message_size = m.group(3)
-            producers = m.group(4)
-            rate = m.group(5)
-            data['file'] = m.group(6)
-        else:
-            print("Something is not right with the file names!")
+        name = "{driver}-{workload}".format(driver=data['driver'], workload=data['workload'])
+        if name in benchmark_names:
+            print(f"Duplicate benchmark found: {name} in file {file}")
             exit(-1)
```

Comment:
Does this not prevent us charting successive runs of the same workload? I'd rather we have an option to allow this.

Reply:
I agree with you, but I'm not really sure where this format is coming from. Benchmark outputs the files in [this format](https://github.com/redpanda-data/openmessaging-benchmark/blob/cde313756faa1ebd0114a35d6fc3b09198559738/benchmark-framework/src/main/java/io/openmessaging/benchmark/Benchmark.java#L159)

```
String.format("%s-%s-%s.json", workloadName, driverConfiguration.name,
                                dateFormat.format(new Date()));
```

When you run this, do you pass a custom file name in this format? 


A couple of was I can fix this 

1. by adding the file name to the check. So,

 benchmark_name = {driver}-{workload}-{file_name}

that will fix it for the case you mentioned, but that'd be awfully long name in the chart label? 

2. Add --label to the benchmark command that embeds this label in the output file along with a new "timestamp" field.

So you can invoke it with something like

benchmark -d {driver} -w {workload} --label "foo" 

and the chart label would be foo-{timestamp_of_run} .

(2) seems clean but that'll diverge a bit from the upstream repo if that is a concern.

Reply:
I've had a check and it seems to work fine for me.